internal/compiler/emitter.go

```diff
@@ -56,11 +57,11 @@ func (e *emitter) getDeclarationTransformers(emitContext *printer.EmitContext, d
 	return []*declarations.DeclarationTransformer{transform}
 }
 
-func getModuleTransformer(emitContext *printer.EmitContext, options *core.CompilerOptions, resolver binder.ReferenceResolver, getEmitModuleFormatOfFile func(file ast.HasFileName) core.ModuleKind) *transformers.Transformer {
-	switch options.GetEmitModuleKind() {
+func getModuleTransformer(ctx context.Context, resolver binder.ReferenceResolver, getEmitModuleFormatOfFile func(file ast.HasFileName) core.ModuleKind) *transformers.Transformer {
+	switch transformers.GetCompilerOptionsFromContext(ctx).GetEmitModuleKind() {
```

Comment:
I feel kinda weird about `context.Context` being a grab back for stuff that was previously nice and static, but I suppose it's fine...